### PR TITLE
Nuke: loading gizmos fixes

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Gizmo product is not offered in Loader as plugin. It is also updating as expected.

## Additional info
Nuke api lib is having new context function for swapping nodes with maintained linked dependencies.

## Testing notes:
1. Load any Gizmo product into scene which is having more then one version. (publish some in advance if you are having none. Two versions are enough)
2. Load the Gizmo into the script and connect it to graph so you are having input and output dependencies.
3. Open Manager of inventory and change version of the loaded gizmo
4. all should work as expected.
5. Use bot loaders > `Load Gizmo - Input Process` and `Load Gizmo`

closes https://github.com/ynput/OpenPype/issues/5501

### CODE REVIEW

File: openpype/hosts/nuke/api/lib.py [hunk 1]
```diff
@@ -2802,7 +2797,7 @@ def find_free_space_to_paste_nodes(
 
 
 @contextlib.contextmanager
-def maintained_selection():
+def maintained_selection(exclude_nodes=None):
```
Comment (Minkiu) [inline openpype/hosts/nuke/api/lib.py:2800]: Missing docstrings.

File: openpype/hosts/nuke/plugins/load/load_gizmo_ip.py [hunk 2]
```diff
@@ -28,7 +29,7 @@ class LoadGizmoInputProcess(load.LoaderPlugin):
 
     families = ["gizmo"]
     representations = ["*"]
-    extensions = {"gizmo"}
+    extensions = {"nk"}
```
Comment (Minkiu) [inline openpype/hosts/nuke/plugins/load/load_gizmo_ip.py:32]: Should we keep both? `nk` and `gizmo`?

**Review by Minkiu — APPROVED** [review]
Haven' tested, code looks good.

**Review by 64qam — APPROVED** [review]

**Review by moonyuet — APPROVED** [review]
Looks good for me with simple test on gizmo publishing and i agree with oscar that there should be docstring for the function.
image

File: openpype/hosts/nuke/plugins/load/load_gizmo_ip.py (hunk 2 above)
Comment (jakubjezek001) [reply openpype/hosts/nuke/plugins/load/load_gizmo_ip.py:32]: only `nk` is the extension. `gizmo` is representation name in our publishing schema.

**Pull request merged**